pwndbg/commands/kmem_trace.py

```
@@ -185,9 +212,12 @@ def _palloc_handler() -> bool:
 
     @staticmethod
     def palloc_handler(sp: pwndbg.dbg_mod.StopPoint) -> bool:
+        inf = pwndbg.dbg.selected_inferior()
+        assert inf
```

Does this make sense? if `inf` is not there, this will crash anyway with TypeError - executing method on NoneType?

---

im a bit torn on this because
1. i believe there is a difference between explicitly documenting an expectation and having it implicit
2. doing this satisfies type correctness hints
3. the assert will give a worse looking backtrace to the user than a TypeError would

---

really, I wish the documentation for these functions
```python
    def selected_inferior(self) -> Process | None:
        """
        The inferior process currently being focused on in this interactive session.
        """
        raise NotImplementedError()

    def selected_thread(self) -> Thread | None:
        """
        The thread currently being focused on in this interactive session.
        """
        raise NotImplementedError()

    def selected_frame(self) -> Frame | None:
        """
        The stack frame currently being focused on in this interactive session.
        """
        raise NotImplementedError()
```
would explicitly say *when* they can return `None` so I can know if that invariant is satisfied when I'm writing some code or not.
@mbrla0 
is `selected_inferior` being non-None equivalent to `@pwndbg.commands.OnlyWhenRunning` ?

---

We should add this docs then; also, the `x=Sth; assert x` is code that executes/its not optimized, so that's why I am a bit hesitant, but we can have it for now too..

---

i think a potentially nice idea to remedy this is to just have selected_inferior() always return `Process`, and if it can't get a valid process it throws an exception instead of returning None.

and then the few places which actually need to check that a valid inferior exists can catch that exception

this way we have both type safety and performance and less verbose code.